parts/k8s/addons/cilium.yaml

```diff
@@ -301,71 +537,181 @@ spec:
         volumeMounts:
         - mountPath: /sys/fs/bpf
           name: bpf-maps
+          mountPropagation: HostToContainer
         - mountPath: /var/run/cilium
           name: cilium-run
-      priorityClassName: system-node-critical
+        resources:
+          requests:
+            cpu: 100m
+            memory: 100Mi
       restartPolicy: Always
       serviceAccount: cilium
       serviceAccountName: cilium
       terminationGracePeriodSeconds: 1
       tolerations:
       - operator: Exists
-      - effect: NoSchedule
-        key: node.kubernetes.io/not-ready
-        operator: Exists
       volumes:
+        # To keep state between restarts / upgrades
       - hostPath:
           path: /var/run/cilium
           type: DirectoryOrCreate
         name: cilium-run
+        # To keep state between restarts / upgrades for bpf maps
       - hostPath:
           path: /sys/fs/bpf
           type: DirectoryOrCreate
         name: bpf-maps
+      # To install cilium cni plugin in the host
       - hostPath:
-          path: /var/run/docker.sock
-          type: Socket
-        name: docker-socket
-      - hostPath:
-          path: /opt/cni/bin
+          path:  /opt/cni/bin
           type: DirectoryOrCreate
         name: cni-path
+        # To install cilium cni configuration in the host
       - hostPath:
           path: /etc/cni/net.d
           type: DirectoryOrCreate
         name: etc-cni-netd
-      - configMap:
-          defaultMode: 420
-          items:
-          - key: etcd-config
-            path: etcd.config
-          name: cilium-config
-        name: etcd-config-path
-      - name: etcd-secrets
-        secret:
-          defaultMode: 420
-          optional: true
-          secretName: cilium-etcd-secrets
+        # To be able to load kernel modules
+      - hostPath:
+          path: /lib/modules
+        name: lib-modules
+        # To access iptables concurrently with other processes (e.g. kube-proxy)
+      - hostPath:
+          path: /run/xtables.lock
+          type: FileOrCreate
+        name: xtables-lock
+      - hostPath:
+          path: /tmp/cilium-bootstrap-time
+          type: FileOrCreate
+        name: cilium-bootstrap-file
+        # To read the clustermesh configuration
       - name: clustermesh-secrets
         secret:
           defaultMode: 420
           optional: true
           secretName: cilium-clustermesh
+        # To read the configuration from the config map
+      - configMap:
+          name: cilium-config
+        name: cilium-config-path
+      - configMap:
+          name: cni-configuration
+        name: cni-configuration
```

Comment:
@denniszielke we are specifying a volume to the `cni-configuration` ConfigMap, but there is no `cni-configuration` ConfigMap defined. Are we supposed to define that in this spec, or does the cilium daemonset spec depend upon the cluster already having that ConfigMap defined?

Reply:
Looks like this was due to `global.cni.configMap=cni-configuration` being set, so it probably means the configmap has to be present...

Reply:
added the configmap.
Do we need to change the config for kubenet ?

Reply:
I don't think there's a kubenet story here. @errordeveloper correct me if I'm wrong...

Reply:
Adding the ConfigMap as you did is fine